src/onegov/org/models/ticket.py

```diff
@@ -376,6 +374,13 @@ def reservations(self):
 
         return tuple(query)
 
+    def has_future_reservation(self):
+        exists = self.session.query(Reservation).filter(
+            Reservation.start > func.now()
+        ).limit(1).all()
+
+        return bool(exists)
```

Comment:
```suggestion
        exists = self.session.query(Reservation).filter(
            Reservation.start > func.now()
        ).exists()

        return self.session.query(exists).scalar()
```
An exists subquery is much more efficient than a query limited to 1 record. (It also avoids the ORM overhead)

But also, shouldn't you filter the reservations to the ones attached to the ticket? Otherwise you will only ever be able to delete tickets if no future reservations exist at all.

Reply:
Very good catch here, this got removed when I refactored. the previous version I had was using `self.reservations` which is what we probably want here.

Reply:
I still can't believe I missed that. Might have to add a test for that